File: .codex-artifacts/operational-case-tracker-20260831/query-contract-status-graph.py
import json
from pathlib import Path

import networkx as nx
from networkx.readwrite import json_graph

graph_path = Path(r"C:\Users\khamis\Documents\fleetifyapp\graphify-out\graph.json")
data = json.loads(graph_path.read_text(encoding="utf-8"))
graph = json_graph.node_link_graph(data, edges="links")
terms = ["contract", "status", "contractstatus"]

scored = []
for node_id, node_data in graph.nodes(data=True):
    label = str(node_data.get("label", "")).lower()
    score = sum(1 for term in terms if term in label)
    source = str(node_data.get("source_file", ""))
    if score and ".archive/" not in source.replace("\\", "/"):
        scored.append((score, graph.degree(node_id), node_id))
scored.sort(reverse=True)
start_nodes = [node_id for _, _, node_id in scored[:3]]

subgraph_nodes = set(start_nodes)
frontier = set(start_nodes)
edges = []
for _ in range(2):
    next_frontier = set()
    for node_id in frontier:
        for neighbor in graph.neighbors(node_id):
            neighbor_source = str(graph.nodes[neighbor].get("source_file", ""))
            if ".archive/" in neighbor_source.replace("\\", "/"):
                continue
            if neighbor not in subgraph_nodes:
                next_frontier.add(neighbor)
                edges.append((node_id, neighbor))
    subgraph_nodes.update(next_frontier)
    frontier = next_frontier

print("START")
for node_id in start_nodes:
    node = graph.nodes[node_id]
    print(node.get("label"), node.get("source_file"), node.get("source_location"))
print("NODES")
ranked = sorted(
    subgraph_nodes,
    key=lambda node_id: (
        sum(term in str(graph.nodes[node_id].get("label", "")).lower() for term in terms),
        graph.degree(node_id),
    ),
    reverse=True,
)
for node_id in ranked[:80]:
    node = graph.nodes[node_id]
    print(
        f"{node.get('label')} | src={node.get('source_file', '')} | "
        f"loc={node.get('source_location', '')}"
    )
print("EDGES")
for source, target in edges[:120]:
    raw = graph[source][target]
    edge = next(iter(raw.values()), {}) if isinstance(graph, nx.MultiGraph) else raw
    print(
        f"{graph.nodes[source].get('label', source)} --{edge.get('relation', '')}--> "
        f"{graph.nodes[target].get('label', target)}"
    )

File: .codex-artifacts/operational-case-tracker-20260831/query-contract-status-graph.mjs
import fs from 'node:fs/promises';

const graphPath = 'C:/Users/khamis/Documents/fleetifyapp/graphify-out/graph.json';
const data = JSON.parse(await fs.readFile(graphPath, 'utf8'));
const terms = ['contract', 'status', 'contractstatus'];
const nodesById = new Map(data.nodes.map((node) => [String(node.id), node]));
const adjacency = new Map([...nodesById.keys()].map((id) => [id, new Set()]));
const edgeByPair = new Map();

for (const edge of data.links ?? []) {
  const source = String(typeof edge.source === 'object' ? edge.source.id : edge.source);
  const target = String(typeof edge.target === 'object' ? edge.target.id : edge.target);
  if (!adjacency.has(source) || !adjacency.has(target)) continue;
  adjacency.get(source).add(target);
  adjacency.get(target).add(source);
  edgeByPair.set(`${source}\u0000${target}`, edge);
  edgeByPair.set(`${target}\u0000${source}`, edge);
}

const isActiveSource = (node) => !String(node.source_file ?? '').replaceAll('\\', '/').includes('.archive/');
const scoreNode = (node) => {
  const label = String(node.label ?? '').toLowerCase();
  return terms.reduce((score, term) => score + Number(label.includes(term)), 0);
};

const starts = data.nodes
  .filter(isActiveSource)
  .map((node) => ({ node, score: scoreNode(node), degree: adjacency.get(String(node.id))?.size ?? 0 }))
  .filter(({ score }) => score > 0)
  .sort((a, b) => b.score - a.score || b.degree - a.degree)
  .slice(0, 3)
  .map(({ node }) => String(node.id));

const visited = new Set(starts);
let frontier = [...starts];
const traversedEdges = [];
for (let depth = 0; depth < 2; depth += 1) {
  const next = [];
  for (const id of frontier) {
    for (const neighbor of adjacency.get(id) ?? []) {
      const neighborNode = nodesById.get(neighbor);
      if (!neighborNode || !isActiveSource(neighborNode) || visited.has(neighbor)) continue;
      visited.add(neighbor);
      next.push(neighbor);
      traversedEdges.push([id, neighbor]);
    }
  }
  frontier = next;
}

const ranked = [...visited].sort((a, b) => {
  const left = nodesById.get(a);
  const right = nodesById.get(b);
  return scoreNode(right) - scoreNode(left)
    || (adjacency.get(b)?.size ?? 0) - (adjacency.get(a)?.size ?? 0);
});

console.log('START');
for (const id of starts) {
  const node = nodesById.get(id);
  console.log(`${node.label} | src=${node.source_file ?? ''} | loc=${node.source_location ?? ''}`);
}
console.log('NODES');
for (const id of ranked.slice(0, 80)) {
  const node = nodesById.get(id);
  console.log(`${node.label} | src=${node.source_file ?? ''} | loc=${node.source_location ?? ''}`);
}
console.log('EDGES');
for (const [source, target] of traversedEdges.slice(0, 120)) {
  const edge = edgeByPair.get(`${source}\u0000${target}`) ?? {};
  console.log(`${nodesById.get(source).label} --${edge.relation ?? ''}--> ${nodesById.get(target).label}`);
}
